arc_solvers/models/qa/multi_choice/qa_multi_choice_max_att.py

```diff
@@ -0,0 +1,247 @@
+                                         .format(choice_output_dim,
+                                                 tensor_2_dim))
+
+        self._matrix_attention_question_to_choice = MatrixAttention(att_question_to_choice)
+
+        self._accuracy = CategoricalAccuracy()
+        self._loss = torch.nn.CrossEntropyLoss()
+
+        initializer(self)
+
+    def forward(self,  # type: ignore
+                question: Dict[str, torch.LongTensor],
+                choices_list: Dict[str, torch.LongTensor],
+                label: torch.IntTensor = None,
+                metadata: List[Dict[str, Any]] = None,
+                ) -> Dict[str, torch.Tensor]:
+        # pylint: disable=arguments-differ
+        """
+        Parameters
+        ----------
+        question : Dict[str, torch.LongTensor]
+            From a ``TextField``
+        choices_list : Dict[str, torch.LongTensor]
+            From a ``List[TextField]``
+        label : torch.IntTensor, optional (default = None)
+            From a ``LabelField``
+
+        Returns
+        -------
+        An output dictionary consisting of:
+
+        label_logits : torch.FloatTensor
+            A tensor of shape ``(batch_size, num_labels)`` representing unnormalised log
+            probabilities of the entailment label.
+        label_probs : torch.FloatTensor
+            A tensor of shape ``(batch_size, num_labels)`` representing probabilities of the
+            entailment label.
+        loss : torch.FloatTensor, optional
+            A scalar loss to be optimised.
+        """
+
+        encoded_choices_aggregated = embed_encode_and_aggregate_list_text_field(choices_list,
+                                                                                self._text_field_embedder,
+                                                                                self._embeddings_dropout,
+                                                                                self._choice_encoder,
+                                                                                self._choice_aggregate)  # # bs, choices, hs
+
+        encoded_question_aggregated = embed_encode_and_aggregate_text_field(question, self._text_field_embedder,
+                                                                               self._embeddings_dropout,
+                                                                               self._question_encoder,
+                                                                               self._question_aggregate)  # bs, hs
+
+        q_to_choices_att = self._matrix_attention_question_to_choice(encoded_question_aggregated.unsqueeze(1),
+                                                                     encoded_choices_aggregated).squeeze()
+
+        label_logits = q_to_choices_att
+        label_probs = torch.nn.functional.softmax(label_logits, dim=-1)
+
+        output_dict = {"label_logits": label_logits, "label_probs": label_probs}
```

Comment:
Having worked with the output of this model, I noticed that the choice with the lowest probability is the selected answer choice, almost as if it's predicting whether an answer choice is *incorrect*. I can't see anything in the code so far that explains this, but I have not yet developed a high-level view of all this.

Reply:
In this repository, the max attention is used as a correct answer. If you have used the model from the other repo, it can be the case as it is shared with the odd-one-out solver which selects the choice which has the smallest similarity to the other three choices (using softmin).